### PR TITLE
Add guide for deleting and cleaning up

### PR DESCRIPTION
**Pull request by killianmuldoon**
Signed-off-by: killianmuldoon <kmuldoon@vmware.com>

Add guide for deleting and cleaning up clusters.

### CODE REVIEW

File: deleting-clusters-and-cleaning-up.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+This delete operation ensures that the cluster and all of the objects that form part of the cluster e.g. Machines, Cluster Infrastructure, are correctly deleted. 
+
+To delete all clusters managed by Cluster API run:
+
+```bash
+kubectl delete clusters --all -A
+```
+
+## Cleaning up resources created by this tutorial
+The management cluster in this tutorial is not managed by Cluster API until [you make it self-hosted](./docker-cluster-self-hosted.md).
```
Comment (killianmuldoon) [inline deleting-clusters-and-cleaning-up.md]: The self hosted doc is linked here which currently doesn't lead anywhere until #19 is merged.

File: deleting-clusters-and-cleaning-up.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+To clean up the management cluster before this step run:
```
Comment (sbueringer) [inline deleting-clusters-and-cleaning-up.md]: I think I know what you mean, but this sounds confusing

Comment (sbueringer) [reply deleting-clusters-and-cleaning-up.md]: The self-hosted thing also moves the management stuff back at the end

So I wonder if we should just simplify this to "this is how you just get rid of everything" (with delete clusters --all)

After they delete the mgmt cluster in l.25 everything (=CAPI stuff) breaks anyway

File: deleting-clusters-and-cleaning-up.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+All clusters created using the Docker infrastructure provider can be managed using kind. To clean up every Kubernetes clusters created in the tutorial - returning your system to the state [before initializing the management cluster](./your-first-cluster.md) run:
+
+```bash
+kind delete clusters --all
+```
+
+Additionally, the container registry created as part of initializing the management cluster should be stopped using:
+```bash
+docker stop kind-registry
+```
```
Comment (sbueringer) [inline deleting-clusters-and-cleaning-up.md]: The registry should not exist, right? (except if folks are using the script from the CAPI repo)

File: deleting-clusters-and-cleaning-up.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+```bash
+kubectl delete cluster docker-cluster-one
+```
+This delete operation ensures that the cluster and all of the objects that form part of the cluster e.g. Machines, Cluster Infrastructure, are correctly deleted. 
+
+To delete all clusters managed by Cluster API run:
+
+```bash
+kubectl delete clusters --all -A
```
Comment (sbueringer) [inline deleting-clusters-and-cleaning-up.md:16]: Does this work cross-namespace? (I somehow remember that delete doesn't work with -A)

Just resolve if I'm wrong :)

Comment (killianmuldoon) [reply deleting-clusters-and-cleaning-up.md:16]: I think this usen't to work, but it does now.

File: deleting-clusters-and-cleaning-up.md (hunk 3 above)
Comment (killianmuldoon) [reply deleting-clusters-and-cleaning-up.md]: You're right - I'll remove this.

File: deleting-clusters-and-cleaning-up.md (hunk 2 above)
Comment (killianmuldoon) [reply deleting-clusters-and-cleaning-up.md]: I could seperate this out into different docs - but the idea here is that these are independent sections, with one for deleting clusters and another for cleaning everything up e.g. if they get into a bad state and want to restart the right approach is `kind delete clusters --all`

File: deleting-clusters-and-cleaning-up.md (hunk 4 above)
Comment (sbueringer) [reply deleting-clusters-and-cleaning-up.md:16]: Maybe new kubectl version - again :D

File: deleting-clusters-and-cleaning-up.md (hunk 2 above)
Comment (sbueringer) [reply deleting-clusters-and-cleaning-up.md]: Yup agree. I'm totally fine with different sections for regular happy-path delete cluster vs. cleanup.

I just wonder if our cleanup should simply by `kind delete clusters --all`. I'm not sure it's necessary to potentially have a step where we first delete the management cluster with kind and then all others.

File: deleting-clusters-and-cleaning-up.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+```
+
+## Cleaning up resources created by this tutorial
+
+All clusters created in this tutorial use the Docker infrastructure provider and can be managed using kind. To clean up every Kubernetes cluster created in the tutorial - returning your system to the state [before initializing the management cluster](./your-first-cluster.md) run:
```
Comment (sbueringer) [inline deleting-clusters-and-cleaning-up.md]:
```suggestion
All clusters created in this tutorial use the Docker infrastructure provider and can be managed using kind. To clean up every Kubernetes cluster created in the tutorial - returning your system to the state [before creating the management cluster](./your-first-cluster.md) run:
```

**Pull request merged**